core/frontend/src/PerModelCategoryVisibility.ts

```diff
@@ -103,27 +113,70 @@ class PerModelCategoryVisibilityOverrides extends SortedArray<PerModelCategoryVi
       return PerModelCategoryVisibility.Override.None;
   }
 
-  public setOverride(modelIds: Id64Arg, categoryIds: Id64Arg, override: PerModelCategoryVisibility.Override): void {
+  /**
+   *
+   * @param perModelCategoryVisibility array of model category visibility overrides
+   * @param viewState Optional param viewState. If no viewState is provided, then the viewState associated with the viewport (used to construct this class) is used.
+   * This optional viewState param is useful for apps which may show multiple iModels with multiple viewStates at once. Passing in a viewState ensures that the subcategories cache for the
+   * iModel associated with the provided viewState is populated as opposed to the iModel associated with the viewport which may or may not be an empty iModel.
+   * @returns void promise
+   */
+  public async setOverrides(perModelCategoryVisibility: PerModelCategoryVisibility.Props[], viewState?: ViewState): Promise<void> {
```

Comment:
You take a ViewState but all you need is an IModelConnection.

Reply:
changed it